### PR TITLE
Fix hoormund bug so we don't ignore errors from fetching tokens

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: handlers/hormuud/handler_test.go [hunk 1]
```diff
@@ -186,9 +186,6 @@ var tokenTestCases = []OutgoingTestCase{
 		MsgText: "Simple Message",
 		MsgURN:  "tel:+250788383383",
 		MockResponses: map[string][]*httpx.MockResponse{
-			"https://smsapi.hormuud.com/api/SendSMS": {
-				httpx.NewMockResponse(400, nil, []byte(`{"error": "invalid token"}`)),
-			},
```
Comment (rowanseymour) [inline handlers/hormuud/handler_test.go:191]: we shouldn't be making this call if fetching the token failed

Comment (norkans7) [reply handlers/hormuud/handler_test.go:191]: Yes

File: handlers/hormuud/handler.go [hunk 2]
```diff
@@ -155,16 +151,18 @@ func (h *handler) FetchToken(ctx context.Context, channel courier.Channel, msg c
 	req.Header.Set("Accept", "application/json")
 
 	resp, respBody, err := h.RequestHTTP(req, clog)
-	if err != nil || resp.StatusCode/100 != 2 {
-		return "", errors.Wrapf(err, "error making token request")
+	if err != nil {
+		return "", courier.ErrConnectionFailed
+	} else if resp.StatusCode/100 != 2 {
+		return "", courier.ErrResponseStatus
```
Comment (rowanseymour) [inline handlers/hormuud/handler.go:157]: this check isn't quite right in so many handlers... maybe we should tweak `RequestHTTP` to return`ErrConnectionFailed` or  `ErrResponseStatus`... is there ever a case where we need to handle a non-200 response differently?

Comment (norkans7) [reply handlers/hormuud/handler.go:157]: We can check on that

**Review by norkans7 — APPROVED** [review]

File: handlers/hormuud/handler.go (hunk 2 above)
Comment (rowanseymour) [reply handlers/hormuud/handler.go:157]: I guess the issue is that some handlers want to parse a 400 response to extract an error code

Comment (rowanseymour) [reply handlers/hormuud/handler.go:157]: Well either way I think you need to do an audit of `RequestHTTP` calls...

**Pull request merged**